cluster/src/test/scala/org/apache/spark/sql/SQLFunctionsTestSuite.scala

```diff
@@ -0,0 +1,2773 @@
+      StructField("numericCol", DecimalType, true),
+      StructField("numeric1Col", DecimalType, true),
+      StructField("doublePrecisionCol", DoubleType, true),
+      StructField("realCol", FloatType, true),
+      StructField("stringCol", StringType, true),
+      StructField("timestampCol", TimestampType, true),
+      StructField("varcharCol", StringType, true),
+      StructField("arrayStringCol", ArrayType(StringType), true),
+      StructField("arrayIntCol", ArrayType(IntegerType), true),
+      StructField("mapCol", MapType(IntegerType, StringType), true),
+      StructField("structCol", StructType(Seq(StructField("c1", StringType, false),
+        StructField("c2", IntegerType, false))), true)
+    )
+
+    val data = Seq(
+      Row(1000, null, null, null, "1234567890abcdefghij",
+        date1, new BigDecimal(66), 2.2, 1.0E8f,
+        1000, 1000, "1234567890abcdefghij", new BigDecimal(100000.0),
+        new BigDecimal(100000.0), 2.2, null, "abcd",
+        time1, "abcd'", null, null, null, null),
+      Row(-10, null, true, null, "ABC@#",
+        current_date, new BigDecimal(-66), 0.0111, -2.225E-307f,
+        -10, 10, "ABC@#", new BigDecimal(-1),
+        new BigDecimal(1), 123.56, 0.089f, "abcd",
+        current_timestamp, "SNAPPY'", Array("abc", "def", "efg"),
+        Array(1, 2, 3), scala.collection.immutable.Map(1 -> "abc"),
+        Row("abc", 123))
+    )
+
+    val someDF = sparkSession.createDataFrame(
+      sparkSession.sparkContext.parallelize(data),
+      StructType(schema)
+    )
+    someDF.createTempView("sparkTable")
+  }
+
+  def dropTables(): Unit = {
+    snc.sql("DROP TABLE IF EXISTS rowTable")
+    snc.sql("DROP TABLE IF EXISTS columnTable")
+    sparkSession.sql("DROP TABLE IF EXISTS sparkTable")
+  }
+
+  protected def getTempDir(dirName: String, onlyOnce: Boolean): String = {
+    var log: File = new File(".")
+    if (onlyOnce) {
+      val logParent = log.getAbsoluteFile.getParentFile.getParentFile
+      if (logParent.list().contains("output.txt")) {
+        log = logParent
+      } else if (logParent.getParentFile.list().contains("output.txt")) {
+        log = logParent.getParentFile
+      }
+    }
+    var dest: String = null
+    dest = log.getCanonicalPath + File.separator + dirName
+    val tempDir: File = new File(dest)
+    if (!tempDir.exists) tempDir.mkdir()
+    tempDir.getAbsolutePath
+  }
+
+
+  private def getSortedFiles(file: File): Array[File] = {
+    file.getParentFile.listFiles.filter(_.getName.startsWith(file.getName)).sortBy { f =>
+      val n = f.getName
+      val i = n.lastIndexOf('.')
+      n.substring(i + 1).toInt
+    }
+  }
+
+  def assertQueryFullResultSet(snc: SnappyContext, sparkQuery: String,
+      snappyQuery: String, numRows: Int,
+      queryNum: String, tableType: String,
+      pw: PrintWriter, sqlContext: SparkSession): Any = {
+
+    var snappyDF = snc.sql(snappyQuery)
+    val snappyQueryFileName = s"Snappy_$queryNum.out"
```

Comment:
Can we compare spark and snappy results without writing to files and reading them back?

Reply:
Added different approach for comparing spark and snappy results. Added code for comparing snappy and spark df.